websites/A/AniKenjiMovie/presence.ts

```diff
@@ -0,0 +1,269 @@
+      const Year = document.querySelector('body > div.box-width > div.player-info > div.player-info-text > div.this-desc-labels > span.this-tag')?.textContent?.trim() || 'N/A'
+      let yearOfMovie = ''
+      const yearRegex = /[Nn]ăm\s*(\d+)/
+      const matchYear = Year.match(yearRegex)
+      if (matchYear && matchYear[1]) {
+        yearOfMovie = matchYear[1]
+      }
+      presenceData.details = `${movieName}`
+      presenceData.state = `Tập ${episodeNumberStr} - ⭐ ${Rating} - 🗓️ ${yearOfMovie}`
+    }
+    else {
+      // Đang ở trang xem phim nhưng chưa phát video hoặc không tìm thấy video
+      const Rating = document.querySelector('body > div.box-width > div.player-info > div.player-info-text > div.this-desc-info > span.this-desc-score')?.textContent?.trim() || 'N/A'
+      const Year = document.querySelector('body > div.box-width > div.player-info > div.player-info-text > div.this-desc-labels > span.this-tag')?.textContent?.trim() || 'N/A'
+      let yearOfMovie = ''
+      const yearRegex = /[Nn]ăm\s*(\d+)/
+      const matchYear = Year.match(yearRegex)
+      if (matchYear && matchYear[1]) {
+        yearOfMovie = matchYear[1]
+      }
+      presenceData.details = `${movieName}`
+      presenceData.state = `Tập ${episodeNumberStr} - ⭐ ${Rating} - 🗓️ ${yearOfMovie}`
+    }
+
+    // Thêm nút xem phim
+    if (showButtons) {
+      presenceData.buttons = [
+        {
+          label: '📺 Xem Phim',
+          url: document.location.href,
+        },
+      ]
+    }
+  }
+  else if (
+    document.querySelector(
+      'body > div.box-width > div.player-info > div.player-info-text > div.title > h2 > a > span',
+    )
+  ) {
+    // Trang chi tiết phim (selector cụ thể)
+    const movieName = document.querySelector('body > div.box-width > div.player-info > div.player-info-text > div.title > h2 > a > span')?.textContent?.trim() || ''
+    const fullTitle = document.querySelector('body > div.box-width > div.player-info > div.player-info-text > div.title > h2')?.textContent?.trim() || ''
+    // Ví dụ: fullTitle = "Tên Phim Gì Đó Chi tiết Tập 15"
+    // 2. Khai báo biến để lưu số tập đã lọc
+    let episodeNumberStr = ''
+    // 3. Dùng Regex để tìm và lọc số tập
+    const regex = /[Tt]ập\s*(\d+)/ // Tìm "Tập" hoặc "tập", theo sau bởi số
+    const match = fullTitle.match(regex)
+    const [
+      showButtons,
+      showTimestamps,
+    ] = await Promise.all([
+      presence.getSetting<boolean>('buttons'),
+      presence.getSetting<boolean>('showtimestamps'),
+    ])
+    // 4. Lấy số từ kết quả khớp (nếu có)
+    if (match && match[1]) {
+      episodeNumberStr = match[1] // Chỉ lấy phần số, ví dụ: "15"
+    }
+    if (iFrameVideo && showTimestamps && !Number.isNaN(duration)) {
+      // Có video từ iframe
+      presenceData.smallImageKey = paused ? Assets.Pause : Assets.Play
+      presenceData.smallImageText = paused ? (await strings).pause : (await strings).play
```

Comment:
There is a lot of duplicate code, such as this. Consider refactoring common variables and logic to reduce the verbosity and code duplication